fluid/fluid.cpp

```diff
@@ -300,6 +308,39 @@ void Fluid::pitch_wheel_sens(int chan, int val)
       channel[chan]->pitchWheelSens(val);
       }
 
+/*
+ * setFromKeyPortamento
+ * requires an input for a default value, usually the TPC
+ */
+void Fluid::setFromKeyPortamento(int chan, int defaultValue) {
+      int ptc = get_cc(chan, PORTAMENTO_CTRL);
+      if (validNote(ptc)) {
+            resetPortamento(chan);
+            fromkey_portamento = ptc; 
+            /*
+            // Assumedly this fixed some sort of bug in FluidSynth2
+            if (!validNote(defaultValue))
+                  defaultValue = ptc;*/
+            }
+      else {
+
+            /* determines and returns fromkey portamento */
+            fromkey_portamento = Channel::INVALID_NOTE;
+
+            if (portamentoTime(chan)) {
+                  /* Portamento when Portamento pedal is On */
+                  /* 'fromkey portamento'is determined from the portamento mode
+                   and the most recent note played (prev_note)*/
+                  if (validNote(defaultValue)) {
+                        fromkey_portamento = ptc;
+                        }
+                  else {
+                        fromkey_portamento = lastNote;
+                        }
```

Comment:
Here `if` and `else` are followed by just a single statement -> no braces needed, they just add clutter IMHO